master/internal/core_experiment.go

```diff
@@ -391,11 +391,16 @@ func (m *Master) parseCreateExperiment(req *apiv1.CreateExperimentRequest, user
 		}
 	} else {
 		var compressErr error
-		modelBytes, compressErr = archive.ToTarGz(filesToArchive(req.ModelDefinition))
-		if compressErr != nil {
-			return nil, config, nil, nil, errors.Wrapf(
-				compressErr, "unable to find compress model definition")
+		if req.ModelDefinition != nil {
+			modelBytes, compressErr = archive.ToTarGz(filesToArchive(req.ModelDefinition))
+			if compressErr != nil {
+				return nil, config, nil, nil, errors.Wrapf(
+					compressErr, "unable to find compress model definition")
+			}
+		} else {
+			modelBytes = []byte{}
```

Comment:
Do we need this line? Does something later care about the distinction between empty slice vs nil slice?

Reply:
Oh ok that makes sense to me

Reply:
@gt2345 question: why not define `modelBytes := []byte{}` above instead of making if condition?